test/distributed/test_nccl.py

```diff
@@ -17,32 +18,36 @@
     TestCase = object  # noqa: F811
 
 
+datatypes = [torch.float, torch.bfloat16] if TEST_WITH_ROCM else [torch.float]
+
 class TestNCCL(TestCase):
 
     @unittest.skipIf(IS_WINDOWS, "NCCL doesn't support Windows")
-    def test_unique_id(self):
+    def test_unique_id(self, device):
         uid = nccl.unique_id()
         self.assertIsInstance(uid, bytes)
         self.assertGreater(len(uid), 1)
 
     @unittest.skipIf(IS_WINDOWS, "NCCL doesn't support Windows")
     @unittest.skipIf(not TEST_MULTIGPU, "only one GPU detected")
-    def test_broadcast(self):
-        expected = torch.FloatTensor(128).uniform_()
+    @dtypes(*datatypes)
+    def test_broadcast(self, device, dtype):
+        expected = torch.zeros(128).uniform_().to(dtype=dtype)
         tensors = [expected.cuda()]
         for device in range(1, torch.cuda.device_count()):
             with torch.cuda.device(device):
-                tensors.append(torch.cuda.FloatTensor(128))
+                tensors.append(torch.zeros(128, dtype=dtype).cuda())
 
         nccl.broadcast(tensors)
         for i in range(torch.cuda.device_count()):
             self.assertEqual(tensors[i], expected)
 
     @unittest.skipIf(IS_WINDOWS, "NCCL doesn't support Windows")
     @unittest.skipIf(not TEST_MULTIGPU, "only one GPU detected")
-    def test_reduce(self):
-        tensors = [torch.FloatTensor(128).uniform_() for i in range(nGPUs)]
-        expected = torch.FloatTensor(128).zero_()
+    @dtypes(*datatypes)
+    def test_reduce(self, device, dtype):
+        tensors = [torch.zeros(128).uniform_().to(dtype=dtype) for i in range(nGPUs)]
```

Comment:
Since `_uniform()` is not defined for CPU bfloat16 tensors, we call `_uniform()` on a float32 tensor and then convert to required dtype.